llvm/lib/CodeGen/MachineCopyPropagation.cpp

```diff
@@ -983,7 +983,7 @@ static bool isBackwardPropagatableCopy(const DestSourcePair &CopyOperands,
   if (MRI.isReserved(Def) || MRI.isReserved(Src))
     return false;
 
-  return CopyOperands.Source->isRenamable() && CopyOperands.Source->isKill();
+  return CopyOperands.Source->isKill();
```

Comment:
So... we know a copy is a copy, so it can't have inherent regalloc restrictions.  But that isn't the only reason something might not be renamable, I think.  See https://reviews.llvm.org/D43042 .

What do you think about fixing the copyPhysReg() API to explicitly pass in the "renamable" bit?